airflow/metrics/statsd_logger.py

```diff
@@ -79,80 +87,84 @@ def __init__(
+
+        if self.metrics_validator.test(full_metric_name):
+            self.statsd.incr(full_metric_name, count, rate)
         return None
 
-    @prepare_stat_with_tags
-    @validate_stat
     def decr(
         self,
-        stat: str,
+        metric_name: str,
         count: int = 1,
         rate: float = 1,
         *,
         tags: dict[str, str] | None = None,
     ) -> None:
         """Decrement stat."""
-        if self.metrics_validator.test(stat):
-            return self.statsd.decr(stat, count, rate)
+        full_metric_name = self.get_name(metric_name, tags)
+        if self.metrics_validator.test(full_metric_name):
+            return self.statsd.decr(full_metric_name, count, rate)
         return None
 
-    @prepare_stat_with_tags
-    @validate_stat
     def gauge(
         self,
-        stat: str,
+        metric_name: str,
         value: int | float,
         rate: float = 1,
         delta: bool = False,
         *,
         tags: dict[str, str] | None = None,
     ) -> None:
         """Gauge stat."""
-        if self.metrics_validator.test(stat):
-            return self.statsd.gauge(stat, value, rate, delta)
+        full_metric_name = self.get_name(metric_name, tags)
+        if self.metrics_validator.test(full_metric_name):
+            return self.statsd.gauge(full_metric_name, value, rate, delta)
         return None
 
-    @prepare_stat_with_tags
-    @validate_stat
     def timing(
         self,
-        stat: str,
-        dt: DeltaType,
+        metric_name: str,
+        dt: DeltaType | None,
         *,
         tags: dict[str, str] | None = None,
     ) -> None:
         """Stats timing."""
-        if self.metrics_validator.test(stat):
-            return self.statsd.timing(stat, dt)
+        full_metric_name = self.get_name(metric_name, tags)
+        if self.metrics_validator.test(full_metric_name):
+            return self.statsd.timing(full_metric_name, dt)
         return None
 
-    @prepare_stat_with_tags
-    @validate_stat
     def timer(
         self,
-        stat: str | None = None,
+        metric_name: str | None = None,
         *args,
         tags: dict[str, str] | None = None,
         **kwargs,
     ) -> TimerProtocol:
         """Timer metric that can be cancelled."""
-        if stat and self.metrics_validator.test(stat):
-            return Timer(self.statsd.timer(stat, *args, **kwargs))
+        full_metric_name = self.get_name(metric_name, tags)
+        if full_metric_name and self.metrics_validator.test(full_metric_name):
+            return Timer(self.statsd.timer(full_metric_name, *args, **kwargs))
         return Timer()
 
+    @prepare_metric_name_with_tags
+    @validate_stat
+    def get_name(self, metric_name: str, tags: Attributes | None = None) -> str:
+        """Get metric name with tags, ensuring no invalid keys are included."""
+        if tags:
+            return f"{metric_name},{','.join(f'{k}={v}' for k, v in tags.items())}"
```

Comment:
Please verify this join.    I seem to remember this should be joined on an underscore, not an equals sign?

Reply:
it is equal sign as the it was before I guess

Reply:
Aright, maybe I'm wrong. :+1: